xilem/src/view/transform.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright 2025 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::marker::PhantomData;
+
+use masonry::Affine;
+use xilem_core::{DynMessage, View, ViewMarker};
+
+use crate::{Pod, ViewCtx, WidgetView};
+
+/// A view which transforms `child`.
+///
+/// The transform can be set using the methods on `Transformable`.
+pub fn transforming<Child, State, Action>(child: Child) -> Transformed<Child, State, Action>
+where
+    Child: WidgetView<State, Action>,
+{
+    Transformed {
+        child,
+        transform: Affine::IDENTITY,
+        phantom: PhantomData,
+    }
+}
+
+pub struct Transformed<V, State, Action> {
+    child: V,
+    transform: Affine,
+    phantom: PhantomData<(State, Action)>,
+}
+
+impl<V, State, Action> ViewMarker for Transformed<V, State, Action> {}
+impl<Child, State, Action> View<State, Action, ViewCtx> for Transformed<Child, State, Action>
+where
+    Child: WidgetView<State, Action>,
+    State: 'static,
+    Action: 'static,
+{
+    type Element = Pod<Child::Widget>;
+    type ViewState = Child::ViewState;
+
+    fn build(&self, ctx: &mut ViewCtx) -> (Self::Element, Self::ViewState) {
+        let (mut child_pod, child_state) = self.child.build(ctx);
+        // TODO: Use a marker identity value to detect this more properly
+        if child_pod.transform != Affine::IDENTITY {
+            panic!("Tried to create a `Transformed` with an already controlled Transform");
+        }
+        child_pod.transform = self.transform;
+        (child_pod, child_state)
+    }
+
+    fn rebuild(
+        &self,
+        prev: &Self,
+        view_state: &mut Self::ViewState,
+        ctx: &mut ViewCtx,
+        mut element: xilem_core::Mut<'_, Self::Element>,
+    ) {
+        if self.transform != prev.transform {
+            element.ctx.set_transform(self.transform);
```

Comment:
I think this breaks or leads do undesired behavior, when composing two `Transformed` views. E.g. when the inner changes, the outer not (as also noted here: https://github.com/linebender/xilem/pull/591#discussion_r1765516837). As noted, also one of the reasons for the somewhat complex implementation of xilem_webs attribute overwrite behavior.

(Noted after writing this, that this may lead to a panic above, but then this still has this issue, if the underlying transform is `Affine::IDENTITY`)

Reply:
Yes... this is exactly the reason for the above panic.

> then this still has this issue, if the underlying transform is Affine::IDENTITY

Yes - `Pod::transform` should be an `Option`, to make this robust.